galaxy_ng/app/tasks/publishing.py

```diff
@@ -0,0 +1,25 @@
+import logging
+
+from pulp_ansible.app.tasks.collections import import_collection
+from pulpcore.plugin.models import ContentArtifact
+
+log = logging.getLogger(__name__)
+
+VERSION_CERTIFIED = "certified"
+
+
+def import_and_auto_approve(artifact_pk, **kwargs):
+    """Import collection version and automatically approve.
+
+    Custom task to call pulp_ansible's import_collection() task
+    then automatically approve collection version so no
+    manual approval action needs to occur.
+
+    Approval currently is a collection version certification flag
+    Approval later will be moving from a staging to an approved repo
+    """
+    import_collection(artifact_pk=artifact_pk, repository_pk=kwargs.get('repository_pk', None))
```

Comment:
Wondering if running a sub task like this needs to use enqueue? And/or task groups? I would kind of expect the outer task to end up enqueue'ing the inner task, but I don't know if that is required here. The only example I see of a task running a subtask is https://github.com/pulp/pulpcore/blob/master/pulpcore/app/tasks/importer.py#L194

Reply:
Using enqueue would mean that this task has to wait until the child task is finished.